### ISSUE TITLE
transfer_keep_alive call of assets pallet should allow to transfer the whole balance when the account existence is guaranteed.

### ISSUE THREAD

**Issue by IkerAlus** [paritytech/polkadot-sdk#4212]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

Currently, the `transfer_keep_alive` call of assets pallet fails with error `BalanceLow` when trying to transfer the entire balance of an asset regardless of the account existence being guaranteed by other means (for example, enough DOT balance). The minimum balance checks for the transferred asset should not matter in this case, the account is going to stay alive in any case. 

### Steps to reproduce

Given an account in Asset Hub with enough DOT balance (or enough balance of any sufficient asset), try to transfer the entire balance of any asset using a `transfer_keep_alive` call.

Couple of examples on Polkadot Asset Hub:

- [Example 1.](https://assethub-polkadot.subscan.io/extrinsic/0xb014b0efd87d84b4c5364bfbd00d3cdd33a4cb0a1370e54e96641102eace5962)

- [Example 2.](https://assethub-polkadot.subscan.io/extrinsic/0xe3efac8730625badd0d9588a91e03e315f9c43c2f4fe54a44744507006f8400d)

**Comment by IkerAlus** [comment]
New occurrence of this bug reported by an external service provider: https://assethub-polkadot.subscan.io/extrinsic/0xf775a2a8353883d46ce774e13ac0f3cc932e01d14399f5eabf37a94f37dd5e14

**Comment by bkchr** [comment]
Yeah the implementation of `transfer_keep_alive` doesn't take into account if the asset is `sufficient` and also doesn't check if there are other `provides`. It also looks like the other way around, having a sufficient asset and calling `transfer_keep_alive` via balances.
